spyder/config/base.py

```diff
@@ -542,50 +542,33 @@ def translate_gettext(x):
 #==============================================================================
 # Mac application utilities
 #==============================================================================
-def running_in_mac_app(pyexec=None):
+def running_in_mac_app(pyexec=sys.executable):
     """
-    Check if Python executable is located inside a standalone Mac app.
+    Check if Spyder is running as a macOS bundle app.
+    Checks for SPYDER_APP environment variable to determine this.
 
-    If no executable is provided, the default will check `sys.executable`, i.e.
-    whether Spyder is running from a standalone Mac app.
-
-    This is important for example for the single_instance option and the
-    interpreter status in the statusbar.
+    If python executable is provided, checks if it is the same as the macOS
+    bundle app environment executable.
     """
-    if pyexec is None:
-        pyexec = sys.executable
-
-    bpath = get_mac_app_bundle_path()
+    # Spyder is macOS app
+    mac_app = os.environ.get('SPYDER_APP') is not None
 
-    if bpath and pyexec == osp.join(bpath, 'Contents/MacOS/python'):
+    if mac_app and pyexec == sys.executable:
```

Comment:
```suggestion
    if sys.platform == 'darwin' and mac_app and pyexec == sys.executable:
```

This check can easily be passed in other platforms just by setting `SPYDER_APP`.